### ISSUE TITLE
[3.0] Custom Normalizer is not called 

### ISSUE THREAD

**Issue by tristanbes** [algolia/search-bundle#144]
Given a customized `Normalizer` (that relies on other `Normalizers`) defined as a service & registered in the Symfony app.
I would  expect the search-bundle to use it and but that's not the case.

```yml
algolia_search:
    indices:
        - name: cv
          class: AppBundle\Entity\Cv
          normalizers:
            - AppBundle\Serializer\Normalizer\CvNormalizer
```

```php
<?php

namespace AppBundle\Serializer\Normalizer;

use AppBundle\Entity\Cv;
use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
use Symfony\Component\Serializer\SerializerAwareInterface;
use Symfony\Component\Serializer\SerializerAwareTrait;

class CvNormalizer implements NormalizerInterface, SerializerAwareInterface
{
    use SerializerAwareTrait;

    public function normalize($object, $format = null, array $context = [])
    {
        /* @var Cv $object */

        if ($format == 'searchableArray') {
            if ($object->getExpiresAt() < new \DateTime("18 months ago")) {
                return [];
            }
        }

        return [
            'id'        => $object->getId(),
            'name'      => $object->getPseudo(),
            'pseudo'    => $object->getPseudo(),
            'slug'      => $object->getSlug(),
            'createdAt' => $this->serializer->normalize($object->getCreatedAt(), $format, $context),
            'updatedAt' => $this->serializer->normalize($object->getUpdatedAt(), $format, $context),
            'user'      => $this->serializer->normalize($object->getUser(), $format, $context),
            'profiles'  => $this->serializer->normalize($object->getProfiles(), $format, $context),
            'game'      => $this->serializer->normalize($object->getGame(), $format, $context),
            'platform'  => $this->serializer->normalize($object->getPlatform(), $format, $context),
        ];
    }

    public function supportsNormalization($data, $format = null)
    {
        return $data instanceof Cv;
    }
}
```

```xml
<service id="cv_normalizer" class="AppBundle\Serializer\Normalizer\CvNormalizer" public="false">
    <tag name="serializer.normalizer" />
</service>
```

## Result 

Default Algolia normalizer is used instead of mine 

![hosted_cloud_search_as_a_service___algolia](https://user-images.githubusercontent.com/346010/33486652-2a646a96-d6ab-11e7-816e-01a9ffb85769.png)

**Comment by tristanbes** [comment]
PS: I don't have any other `Normalizer` that supports this `Cv` class. I have only 1 for this class.

**Comment by julienbourdeau** [comment]
Everything looks correct. Can you dump the configuration [here](https://github.com/algolia/search-bundle/blob/master/src/IndexManager.php#L181) and make sure it's aligned with your yaml?

Can you also double-checked the config applies to the correct environment? 

I know it sounds silly but currently, the only way the default normalizer is used is if no normaliser is configured. It's a default value, not really a fallback if nothing can be used. We could argue this is more a bug that a feature by the way, I'll open a PR.

**Comment by tristanbes** [comment]
```
array:1 [
  0 => "AppBundle\Serializer\Normalizer\CvNormalizer"
]
```
Looks good in the dump, yet still seeing data from "default" normalizer

**Comment by tristanbes** [comment]
Found the bug. Patch incoming

**Comment by julienbourdeau** [comment]
Fixed by #153 
Thanks for investigating this!
